src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -1025,7 +1025,9 @@ namespace AppInstaller::Repository
         // Get the details again by name from the source list because SaveMetadata only updates the LastUpdateTime
         // if the details came from the same instance of the list that's being saved.
         // Some sources that do not need updating like the Installed source, do not have Name values.
-        if (!details.Name.empty())
+        // Restricted sources don't have full functionality
+        if (!details.Name.empty() &&
+            !details.Restricted)

```

Comment:
Problem being addressed is: not all sources that can be opened appear in the sourcelist (specifically the one source that is marked as Restricted right now). Constraint is: I think we do want to fail in some cases if the source is not in the source list any more - i.e. scenario: App caller gets list of sources on app open, user removes a source from the command line, app caller eventually tries to Connect to the removed source but it's no longer in the source list. Decision has to be made here whether to fail or succeed. If succeed is ok, then this code can just continue to do the Open even if it doesn't find it in the list. If it needs to fail, then we need a way of knowing whether it's expected that the source isn't in the source list. Right now Restricted sources are expected to not be in the source list, and not need to be updated..

Reply:
Yes. sreadingMSFT said it right. Since winget list should not display Restricted source, it is not available in SourceList and hence, it fails there without this change.